src/Zenith.sol

```diff
@@ -129,19 +100,14 @@ contract Zenith is Passage {
     /// @param v - the v component of the Sequencer's ECSDA signature over the block header.
     /// @param r - the r component of the Sequencer's ECSDA signature over the block header.
     /// @param s - the s component of the Sequencer's ECSDA signature over the block header.
-    /// @custom:reverts BadSequence if the sequence number is not the next block for the given rollup chainId.
-    /// @custom:reverts BlockExpired if the confirmBy time has passed.
+    /// @custom:reverts IncorrectHostBlock if the hostBlockNumber does not match the current block.
     /// @custom:reverts BadSignature if the signer is not a permissioned sequencer,
     ///                 OR if the signature provided commits to a different header.
     /// @custom:reverts OneRollupBlockPerHostBlock if attempting to submit a second rollup block within one host block.
     /// @custom:emits BlockSubmitted if the block is successfully submitted.
     function submitBlock(BlockHeader memory header, uint8 v, bytes32 r, bytes32 s, bytes calldata) external {
-        // assert that the sequence number is valid and increment it
-        uint256 _nextSequence = incrementSequence(header.rollupChainId);
-        if (_nextSequence != header.sequence) revert BadSequence(_nextSequence);
-
-        // assert that confirmBy time has not passed
-        if (block.timestamp > header.confirmBy) revert BlockExpired();
+        // assert that the host block number is valid
+        if (block.number != header.hostBlockNumber) revert IncorrectHostBlock();
```

Comment:
note @prestwich : i realized while implementing that this line is going to mess up default gas estimation for submitting blocks. during gas estimation, `header.hostBlockNumber` should be `current block.number + 1`  (e.g. the next block that will mine). 

i think Builders would need to override checks that the transaction will fail before submitting, which is not ideal

Reply:
within `eth_call` and `eth_estimateGas`, `block.number` is the next (unmined) block number unless you tell it otherwise

Reply:
you can see this behavior here:
https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-eth-api/src/helpers/pending_block.rs#L79-L82

Reply:
slay